### PR TITLE
[bitnami/grafana] add datasources.secretDefinition option

### PR DESCRIPTION
**Pull request by rptaylor**
Signed-off-by: Ryan Taylor <1686627+rptaylor@users.noreply.github.com>

### Description of the change

Add a new option to define datasources, with a secret defined as a Helm value.

### Benefits

Users can define data sources directly via Helm values, instead of having to create and manage an additional extraneous YAML file with the secret definition.  This way all required configuration for the Helm release can be managed cohesively in one place. For some users this will be equally or more secure (for example we deploy Helm releases using Ansible which makes it easy to encrypt values, more so than a plain file in git) while other users who do not have sensitive datasources anyway can enjoy more convenience.

### Possible drawbacks
Users could choose to define sensitive data sources in a values file, but whether this is insecure or not depends on their other practices. In any case it is users' own choice and responsibility to store their sensitive data appropriately.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by rptaylor** [comment]
I tested this with these values:
```
datasources:
  secretDefinition:
    apiVersion: 1
    datasources:
    - name: LocalPrometheus
      type: prometheus
      url: http://kube-prometheus-prometheus.kube-prometheus.svc:9090
      isDefault: true
```
then I check the grafana page and the data source showed up as expected.

**Comment by carrodher** [comment]
Thanks for your contribution, could you please resolve the conflicts?

**Comment by rptaylor** [comment]
@carrodher conflicts resolved and the tests are passing.

**Comment by rptaylor** [comment]
There was another merge conflict due to the chart version getting bumped but I fixed it again.

File: bitnami/grafana/values.yaml [hunk 1]
```diff
@@ -235,12 +235,21 @@ dashboardsProvider:
...
+##   apiVersion: 1
+##   datasources:
+##   - name: Prometheus
+##     type: prometheus
+##     url: http://prometheus-prometheus-server
+##     access: proxy
+##     isDefault: true
 datasources:
   secretName: ""
+  secretDefinition: {}
```
Comment (carrodher) [inline bitnami/grafana/values.yaml:252]:
```suggestion
## Import datasources from an externally-managed secret, or a secret definition set via Helm values.
##
datasources:
  ## @param datasources.secretName The name of an externally-managed secret containing custom datasource files.
  ##
  secretName: ""
  ## @param datasources.secretDefinition The contents of a secret defining a custom datasource file. Only used if datasources.secretName is empty or not defined.
  ## Example:
  ## secretDefinition:
  ##   apiVersion: 1
  ##   datasources:
  ##   - name: Prometheus
  ##     type: prometheus
  ##     url: http://prometheus-prometheus-server
  ##     access: proxy
  ##     isDefault: true
  ##
  secretDefinition: {}
```

**Review by carrodher — APPROVED** [review]
LGTM, thanks

**Pull request merged**